helpers.php

```diff
@@ -11,34 +11,50 @@
 function authorstatsReadJSON() 
 {
     $json = new JSON(JSON_LOOSE_TYPE);
-    $file = @file_get_contents(DOKU_PLUGIN."authorstats/authorstats.json");
+    $file = @file_get_contents(DOKU_PLUGIN."authorstats/data/authorstats.json");
     if(!$file) return Array();
     return $json->decode($file);
 }
 
 // Save the statistics into the JSON file
 function authorstatsSaveJSON($authors) 
 {
+    authorstatsCreateDirIfMissing("data");
     $json = new JSON();
     $json = $json->encode($authors);
-    file_put_contents(DOKU_PLUGIN."authorstats/authorstats.json", $json); 
+    file_put_contents(DOKU_PLUGIN."authorstats/data/authorstats.json", $json); 
 }
 
 // Read the saved statistics for user from the JSON file
 function authorstatsReadUserJSON($loginname)
 {
     $json = new JSON(JSON_LOOSE_TYPE);
-    $file = @file_get_contents(DOKU_PLUGIN."authorstats/".$loginname.".json");
+    $file = @file_get_contents(DOKU_PLUGIN."authorstats/data/".$loginname.".json");
     if(!$file) return Array();
     return $json->decode($file);
 }
 
 // Save the statistics of user into the JSON file
 function authorstatsSaveUserJSON($loginname, $pages)
 {
+    authorstatsCreateDirIfMissing("data");
     $json = new JSON();
     $json = $json->encode($pages);
-    file_put_contents(DOKU_PLUGIN."authorstats/".$loginname.".json", $json);
+    file_put_contents(DOKU_PLUGIN."authorstats/data/".$loginname.".json", $json);
+}
+
+// Creat directory if missing
+function authorstatsCreateDirIfMissing($folder)
+{
+    $path = DOKU_PLUGIN."authorstats/$folder";
+    if (!file_exists($path))
+    {
+        mkdir($path, 0755);
+    }
+    else if (!is_dir($path))
+    {
+        // log or print an error
```

Comment:
Do we want to actually log an error here?

Reply:
I wasn't sure how doku likes to log their error messages, so I left the comment as it's placeholder.  The only way that condition will be met is if a file called "data" exists.  On Linux you can't have a file and folder with the same name.  I figure it would be unlikely since its a subdir internal to the plugin.

Reply:
I removed this block from the code.